### PR TITLE
feat(app): add sign up flow

### PR DESCRIPTION
**Pull request by pedronauck**
This pull request has the entire SignUp flow created, also the logic for public and private routes.

### CODE REVIEW

File: packages/app/src/systems/Account/machines/accountsMachine.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+type MachineServices = {
+  fetchAccounts: {
+    data: Maybe<Account[]>;
+  };
+};
+
+export const accountsMachine = createMachine(
+  {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-imports
+    tsTypes: {} as import("./accountsMachine.typegen").Typegen0,
```
Comment (luizstacio) [inline packages/app/src/systems/Account/machines/accountsMachine.tsx:23]: This is not pretty. But it is a huge improvement for xstate; really nice.

File: packages/app/src/systems/SignUp/routes.tsx [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
+    <Outlet />
+  </PublicRoute>
+);
 
 export const signUpRoutes = (
-  <Route path={`${Pages.signUp}/*`} element={<WelcomeScreen />}>
-    <Route index element={<Navigate to={Pages.signUp} />} />
+  <Route path={`${Pages.signUp}/*`} element={wrapper}>
+    <Route index element={<Navigate to={Pages["signUp.welcome"]} />} />
+    <Route path={Pages["signUp.welcome"]} element={<WelcomeScreen />} />
```
Comment (luizstacio) [inline packages/app/src/systems/SignUp/routes.tsx:16]: I think we should change the `Pages["signUp.welcome"]` to `Pages.signUpWelcome`

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/app/src/systems/Core/components/PrivateRoute/PrivateRoute.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+type PrivateRouteProps = {
+  children: ReactNode;
+};
+
+export function PrivateRoute({ children }: PrivateRouteProps) {
+  const { accounts, isLoading } = useAccounts();
+  if (!isLoading && !accounts?.length) {
+    return <Navigate to="/sign-up" replace />;
+  }
+  return <>{children}</>;
```
[the remark this answers is not in the document]
Comment (pedronauck) [reply packages/app/src/systems/Core/components/PrivateRoute/PrivateRoute.tsx:15]: ✅

File: packages/app/src/systems/SignUp/routes.tsx (hunk 2 above)
Comment (pedronauck) [reply packages/app/src/systems/SignUp/routes.tsx:16]: ✅

File: packages/app/src/systems/Account/machines/accountsMachine.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (pedronauck) [reply packages/app/src/systems/Account/machines/accountsMachine.tsx:23]: In fact, we don't have any README yet, I guess we can create it and add this document there. I'll create an issue for this!

File: packages/app/src/systems/Core/components/PrivateRoute/PrivateRoute.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (pedronauck) [reply packages/app/src/systems/Core/components/PrivateRoute/PrivateRoute.tsx:15]: Returning just `children` generates an error on Typescript
Screen Shot 2022-09-01 at 5 09 37 PM

File: packages/app/src/systems/Account/machines/accountsMachine.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (luizstacio) [reply packages/app/src/systems/Account/machines/accountsMachine.tsx:23]: Just a comment don't need to be on the readme is better to be contextual in this case

**Pull request merged**